torch/_export/__init__.py

```diff
@@ -118,8 +118,16 @@ def tree_zip(combined_args, dynamic_shapes):
                     UserErrorType.INVALID_INPUT,
                     f"Expected {dynamic_shapes} to have {len(combined_args)} items",
                 )
-            for k, shape in dynamic_shapes.items():
-                yield from tree_zip(combined_args[k], shape)
+
```

Comment:
I'm not sure this is a good idea to do recursively. It will be very confusing.

Instead, look below where `combined_args` is created and then matched with `dynamic_shapes` with the top-level call to `tree_zip`. You should do the adjustment there. Specifically, if `dynamic_shapes` is a tuple then you do `combined_args = tuple(combined_args.values())` and that's it.